R/template-utils.R

```diff
@@ -0,0 +1,47 @@
+#' Generate display string
+#' 
+#' Populates passed template with values from display_variables after verifying.
+#' 
+#' @param template string template containg variables to insert surrounded by {}
+#' @param display_variables named list with values to insert where name is equal to name in template
+#' @param ID string containing ID used in warning generation
+#' 
+#' @return string which contains the populated template
+generate_display <- function(template, display_variables, ID){
+  # Check if appropriate variables were past for a template
+  detected_variables <- extract_variables(template)
+  for (single_variable in detected_variables) {
+    if(!(single_variable %in% names(display_variables))){
+      stop(glue::glue('{single_variable} Is missing from the passed display'))
```

Comment:
This should probably collect all the misssing variables and then stop execution. Otherwise it could get annoying running it one at a time.